gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/client/GafferClient.java

```diff
@@ -25,23 +27,20 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Repository;
 import uk.gov.gchq.gaffer.common.model.v1.Gaffer;
-import uk.gov.gchq.gaffer.common.model.v1.GafferList;
-import uk.gov.gchq.gaffer.common.util.CommonUtil;
 import uk.gov.gchq.gaffer.gaas.exception.GaaSRestApiException;
+import uk.gov.gchq.gaffer.gaas.handlers.DeploymentHandler;
 import uk.gov.gchq.gaffer.gaas.model.GaaSGraph;
 import uk.gov.gchq.gaffer.gaas.model.GraphUrl;
+
 import java.util.List;
 import java.util.stream.Collectors;
-import static uk.gov.gchq.gaffer.common.util.Constants.GROUP;
-import static uk.gov.gchq.gaffer.common.util.Constants.PLURAL;
-import static uk.gov.gchq.gaffer.common.util.Constants.VERSION;
-import static uk.gov.gchq.gaffer.gaas.factories.GaaSGraphsFactory.from;
+
 import static uk.gov.gchq.gaffer.gaas.factories.GaaSRestExceptionFactory.from;
-import static uk.gov.gchq.gaffer.gaas.util.Properties.NAMESPACE;
 
 @Repository
-public class CRDClient {
-    private static final Logger LOGGER = LoggerFactory.getLogger(CRDClient.class);
+public class GafferClient {
```

Comment:
Just so I am getting my head round this, the way it used to work was that the rest api would make a custom resource definition (crd) request which basically says make me this custom Gaffer. Then the CRDClient would be listening for that request and actually go about using kubernetes to create it? But now, the CRDClient is all binned and the rest api itself calls kubernetes to create the Gaffers.

I know it's a little late to be asking questions like this but just wondering why this move was decided? Is it as secure? And if it is the best way to do it, why was it done with a CRD Client in the first place? If I remember the initial Kai did something more akin to this but then it was reworked to use CRD Client and now it seems to be going back again to a slightly simpler form (simpler is probably better!).

Sorry for such a big question!

Reply:
And to do this it is using `io.fabric8.kubernetes.client.KubernetesClient`? Is it official? How widely used / trusted is this? And if it is not official, can the official api not be used to do this?

Reply:
The fabric8 kubernetesClient is widely used and is still supported/actively updated. Unfortunately the official API is rather limited in its functionality so would not support all the things we want to do.

Reply:
Combining the UI and CRD makes the whole thing less complex and easier to install